test/unit/services/finance/billing_service_test.rb

```diff
@@ -28,23 +28,23 @@ class Finance::BillingServiceTest < ActionDispatch::IntegrationTest
     end
   end
 
-  test 'triggers billing' do
+  test 'does not trigger billing without a buyer' do
     Sidekiq::Testing.inline! do
       now = Time.utc(2018, 1, 16, 8)
       billing_options = { only: [@provider.id], now: now, skip_notifications: true }
-      Finance::BillingStrategy.expects(:daily).with(billing_options).returns(mock_billing_success(now, @provider))
-      assert Finance::BillingService.call!(@provider.id, now: now, skip_notifications: true)
+      Finance::BillingStrategy.expects(:daily).never
+      Finance::BillingService.call!(@provider.id, now: now, skip_notifications: true)
     end
   end
 
   test 'lock remains if execution fails' do
     now = Time.utc(2018, 1, 16, 8)
     Finance::BillingStrategy.expects(:daily).raises RuntimeError, 'random failure'
     assert_raises RuntimeError do
-      Finance::BillingService.call!(@provider.id, now: now)
+      Finance::BillingService.call!(@provider.id, now: now, provider_account_id: master_account.id)
```

Comment:
```suggestion
      Finance::BillingService.call!(@provider.id, { now: now, provider_account_id: master_account.id })
```

In preparation for Ruby 2.7 / Ruby 3 upgrade